### PR TITLE
keyboards/anavi: Add ANAVI Macro Pad 10

### PR DESCRIPTION
**Pull request by leon-anavi**
ANAVI Macro Pad 10 is a mini hot-swappable mechanical keyboard with a clickable rotary encoder, 9 Cherry MX compatible switches, translucent key caps, USB-C, RP2040 microcontroller, backlighting and under lighting.

This commits adds initial QMK support for ANAVI Macro Pad 10 with:
- Default keymap
- Rotary encoder for controlling the volume
- WS2812B RGB underlighting

Several hardware features of ANAVI Macro Pad 10 are not supported yet: backlighting, neopixel on the XIAO RP2040 module and the click of the encoder (aka to use the encoder as a key).

Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>

## Description

Add a new keyboard: ANAVI Macro Pad 10 with Raspberry Pi RP2040 MCU, This is a mini hot-swappable mechanical keyboard with a clickable rotary encoder, 9 Cherry MX compatible switches, translucent key caps, USB-C, backlighting and under lighting.

This commits adds initial QMK support for ANAVI Macro Pad 10 with:
- Default keymap
- Rotary encoder for controlling the volume
- WS2812B RGB underlighting

Several hardware features of ANAVI Macro Pad 10 are not supported yet: backlighting, neopixel on the XIAO RP2040 module and the click of the encoder (aka to use the encoder as a key).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/anavi/macropad10/info.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    },
+    "rgblight": {
+        "pin": "GP3",
+        "led_count": 4,
+	"hue_steps": 10,
+	"saturation_steps": 17,
+        "brightness_steps": 17,
+	"max_brightness": 255,
+	"animations": {
+            "all": true
```
Comment (drashna) [inline keyboards/anavi/macropad10/info.json:30]: Could you explicitly define which animations are enabled? 
```suggestion
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/anavi/macropad10/info.json (hunk 1 above)
Comment (leon-anavi) [reply keyboards/anavi/macropad10/info.json]: Thank you for the feedback. I have modified `info.json` and forced pushed the updated commit. Thanks, Leon

File: keyboards/anavi/macropad10/rules.mk [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# MCU name
+MCU = RP2040
+# Bootloader selection
+BOOTLOADER = rp2040
+
```
Comment (fauxpark) [inline keyboards/anavi/macropad10/rules.mk:5]:
```suggestion
```
These are already in info.json.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/anavi/macropad10/rules.mk (hunk 2 above)
Comment (leon-anavi) [reply keyboards/anavi/macropad10/rules.mk]: Thank you for the code review. I have modified `rules.mk` and forced pushed the updated commit. Thanks, Leon

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**